testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToHudi.java

```diff
@@ -0,0 +1,448 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hive;
+
+import io.trino.tempto.BeforeTestWithContext;
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.assertions.QueryAssert;
+import io.trino.tempto.query.QueryResult;
+import org.testng.annotations.Test;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.hive.util.HiveUtil.HUDI_TABLE_TYPE_COPY_ON_WRITE;
+import static io.trino.plugin.hive.util.HiveUtil.HUDI_TABLE_TYPE_MERGE_ON_READ;
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertQueryFailure;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.tests.product.TestGroups.HIVE_HUDI_REDIRECTIONS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.utils.QueryExecutors.onHudi;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TestHiveRedirectionToHudi
+        extends ProductTest
+{
+    private String bucketName;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        bucketName = requireNonNull(System.getenv().getOrDefault("S3_BUCKET", "trino-ci-test"), "Environment variable not set: S3_BUCKET");
+    }
+
+    private record TestTableName(String catalog, String schema, String table)
```

Comment:
Introducing a helper method `hudiTableName` looks enough in my opinion.

Reply:
it's not only for hudi's table names but it also uses for both hive and hudi.

Reply:
I know. Please take a look at `TestIcebergSparkCompatibility`. It contains `trinoTableName` & `sparkTableName`.

Reply:
sorry, it's irrelevant sample since there are no usage with/without schema and catalog name either of those as single object.

Reply:
replaced for helper methods